pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,461 @@
+syntax = "proto3";
+
+package v1alpha1;
+
+import "k8s.io/api/core/v1/generated.proto";
+import "k8s.io/api/autoscaling/v1/generated.proto";
+import "k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto";
+import "github.com/gogo/protobuf/protobuf/google/protobuf/wrappers.proto";
+
+message IstioInstall {
+    InstallSpec spec = 3;
+    InstallStatus status = 4;
+}
+
+// InstallSpec defines the desired state of IstioInstaller.
+// The spec is a used to define a customization of the default values that are supplied with an Istio charts release.
+// It is grouped at the top level by feature, where behavior of Istio functional areas is specified.
+// Each feature contains components, where k8s resource level defaults can be overridden.
+// Because the spec is a customization API, specifying an empty InstallSpec results in a default Istio installation.
+message InstallSpec {
+    // Prefix added to all namespaces for any installed component.
+    string default_namespace_prefix = 5;
+
+    // Selection and configuration of core Istio features.
+    TrafficManagementFeatureSpec traffic_management = 25;
+    PolicyFeatureSpec policy = 26;
+    TelemetryFeatureSpec telemetry = 27;
+    SecurityFeatureSpec security = 28;
+    ConfigManagementFeatureSpec config_management = 29;
+
+    // Ingress/egress gateway installation and configuration.
+    repeated IngressGatewaySpec ingress_gateway = 30;
+    repeated EgressGatewaySpec egress_gateway = 31;
+
+    // List of external operators that will be installed (but not managed)
+    // by the IstioInstaller e.g. prometheus, grafana, kiali.
+    repeated OperatorSpec external_operators = 50;
```

Comment:
I think this is a slippery slope.  I don't think we should be doing anything with external operators, but should instead have some validation that rejects CRs if they would require a CRD supported by an external operator that hasn't been installed.  Using something like OLM would be a better option here.

Reply:
What is OLM? This is just an install passthrough, the use case is to just install an external operator (e.g. prometheus for demo) and defer control of it to that sub-operator. There's no interaction with the Istio operator, it's just a convenience so we don't keep maintaining add on config in Istio for demo purposes.

Reply:
So the external operator spec here is just to keep track of sub-operators, Istio operator won't have logic to handle CR of sub-operators.

Reply:
Operators should not be tracking other operators, that is the job of other tools such as the Operator Lifecycle Manager.  The operator should only be concerned with Custom Resources monitored by any dependent operators.

Reply:
we may want to leave out external_operator since it is confusing why user needs to specify istio operator isn't managing it.

Reply:
I think for simplicity, we should just focus on managing the installation of the istio control plane.  Longer term, I suspect downstream users/adopters might create custom operators which manage and installation that includes istio, kiali, etc.

If the control plan has a hard dependency on prometheus, then, as @knrc mentioned, we should rely on something else to manage the operator dependencies, and the istio operator should be installing whatever CR is required by that operator.  If no operator management system is being used, it would fall upon the user to ensure that the required operators are installed.

Reply:
Let me emphasize again, Istio operator will not manage the external operator, only bootstrap it into the cluster and leave it alone. I've changed the text to make that more clear. 
The reason we need this is for simple installs of demos that include things like prometheus/grafana etc. 
Istio is getting out of the business of managing those but we still need a "one-line" install that includes those components. We don't want to require users to install OLM to do that, if they just want to kick the tires of Istio.

Reply:
If istio installs it, do we expect other operators to manage it?  Or would it leave as unmanaged?

It seems this is only used for demo and quick start?  else, folks would use external operator to install the component so it can be managed.

Reply:
ok SG thanks

Reply:
I still think this is a bad idea, and saying it's just for demos, I think makes it worse.  If this is for demos, then I don't think it's too big a hurdle to document that somebody doing a demo will have to install other operators if they need/want third-party features.

I think something like this also emphasizes the use case for some sort of built-in extension mechanism or hook that gives users access to setting non-standard or less common configuration details. This might look something like:

```yaml
trafficManagement:
  ...
policy:
  ...
...
rawValues:
  kiali:
    enabled: true
```

This provides users with a concise set of features to configure, while allowing users to bring their own chart sets and still be able to configure them.  In the case above, say kiali is no longer part of the standard chart set, but is included in some third-party distribution, then that user would have the ability to configure those charts, even though there's not direct support in the CRD itself (well, I guess `rawValues` would be direct support).

Reply:
Regarding item two, I think it's pretty simple, the operator would install a CR  managed by another operator.  If the CRD isn't available, the install fails.  Hopefully, the error reporting is good enough so the user sees something like, `install failed: kiali install failed: kiali.io/v1alpha1,Kind=kiali CRD not found`.

Reply:
Ugh, I think I may now have to put it back exactly for that reason. We never want the install to fail (even a demo) because the CRD is not available and I think the only way to ensure it is to have the operator create that before setting the CR.

Reply:
I don't think this is really an issue.  As the user will have to install the istio operator itself, those additional operator installations can just be part of that installation process, e.g. https://github.com/istio/operator/tree/master/deploy.  (Also, if using OLM, those dependencies can be defined and OLM will take care of installing those dependencies.)

Reply:
We need a one line demo install option - I think that's an absolutely essential requirement. But I'm going to leave this out of this PR in the interests of moving ahead and have created https://github.com/istio/istio/issues/14534 to track this.